lib/Slack_web_api.js

```diff
@@ -127,10 +127,7 @@ module.exports = function(bot, config) {
      * @param {function} cb A NodeJS-style callback
      */
     slack_api.callAPI = function(command, data, cb, multipart) {
-        if (!data.token) {
-            data.token = config.token;
-        }
-
+        data.token = data.token ? data.token : config.token;
```

Comment:
If data is defined on the same scope and in the same line, JS interpreter will throw an error. You're right. But the thing is `data` variable is defined before this line and therefore will not throw an error if data.token is undefined.

Example:

```javascript
> var data = {}
< undefined
> data.token = data.token || 3;
< 3
> data.token
< 3
```